test/test_integration.py

```diff
@@ -54,58 +56,129 @@ def tearDown(self):
     def test_integration(self):
 
         # Create tenant
-        # TODO WORKING WITH ISSUES*! [ValueError: Invalid value for `api_key`, must not be `None`]
+        random_api_key = ''.join(choice(ascii_lowercase) for i in range(4))
+        random_api_secret = ''.join(choice(ascii_lowercase) for i in range(5))
         tenant = killbill.api.TenantApi()
-        body = Tenant(api_key='bob_5', api_secret='lazar_5')
+        body = Tenant(api_key=random_api_key, api_secret=random_api_secret)
         tenant.create_tenant(body, 'test')
 
         # Get tenant
-        # TODO NOT WORKING! [ApiException: (401) Reason: Unauthorized] or [ValueError: Invalid value for `api_secret`, must not be `None`]
-        body = Tenant(api_key='bob')
-        tenant.get_tenant_by_api_key(body)
+        tenant = killbill.api.TenantApi()
+        tenant = tenant.get_tenant_by_api_key(api_key=random_api_key)
+        self.assertIsNotNone(tenant.tenant_id)
+        self.assertEqual(random_api_key, tenant.api_key)
+
 
         # Upload XML catalog/Fetch XML catalog
-        # TODO NOT WORKING! [ApiException: (404) Reason: Not Found]
         catalog = killbill.api.CatalogApi()
-        xml_catalog = open("/home/jfrodriguez/Documents/KillBill-Docs/SpyCarBasic.xml", "r+").read()
-        catalog.upload_catalog_xml(xml_catalog, 'test', 'bob', 'lazar')
+        xml_catalog = open("../resources/SpyCarBasic.xml", "r+").read()
+        catalog.upload_catalog_xml(xml_catalog, 'test', random_api_key, random_api_secret)
+
+        # Get catalog
+        catalog = killbill.api.CatalogApi()
+        catalog = catalog.get_catalog_xml(random_api_key, random_api_secret)
+        self.assertIsNotNone(catalog)
```

Comment:
@wwjfrodriguez Could you at least verify with your eyes that  catalog xml returned looks fine?

Reply:
@sbrossie yes, I verified with `print catalog` and its shown in the console.